src/inmanta/data/model.py

```diff
@@ -475,3 +475,13 @@ class Agent(BaseModel):
     process_name: Optional[str]
     unpause_on_resume: Optional[bool]
     status: const.AgentStatus
+
+
+class AgentProcessDetails(BaseModel):
+    sid: uuid.UUID
+    hostname: str
+    environment: uuid.UUID
+    first_seen: Optional[datetime.datetime]
+    last_seen: Optional[datetime.datetime]
+    expired: Optional[datetime.datetime]
+    state: Optional[Union[Dict[str, List[str]], Dict[str, str], Dict[str, float], str]]
```

Comment:
Using `ReportReturn` here causes an import loop

Reply:
Since the `ReportReturn` is only used for type checking, you can use the trick with the `TYPE_CHECKING` variable to get out of the import loop (See: https://docs.python.org/3/library/typing.html#typing.TYPE_CHECKING)